src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/activity/Activity.java

```diff
@@ -1,42 +1,93 @@
 package com.wikia.webdriver.pageobjectsfactory.componentobject.activity;
 
 import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+
 import com.wikia.webdriver.pageobjectsfactory.pageobject.UserProfilePage;
-import com.wikia.webdriver.pageobjectsfactory.pageobject.article.ArticlePageObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.diffpage.DiffPagePageObject;
+import lombok.Getter;
 import org.openqa.selenium.By;
-import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 
+import javax.swing.text.html.HTML;
+import java.util.Arrays;
+
 
 public class Activity extends BasePageObject {
 
-  private WebElement userLink;
-  private WebElement titleLink;
+  private By userLink = By.cssSelector("cite > span > a");
+  private By titleLink = By.cssSelector("a.title");
+  private By diffLink = By.cssSelector("cite > a.activityfeed-diff");
+  private By wallOwner = By.cssSelector(".wall-owner");
+  private By wallThreadAuthor = By.cssSelector(".wallfeed tr:first-child .real-name");
+  private By wallThreadContent = By.cssSelector(".wallfeed tr:first-child p:nth-child(2)");
+  private By description = By.cssSelector("table");
+
+  private WebElement entry;
 
-  public Activity(WebDriver driver, WebElement parentElement) {
-    super();
+  @Getter
+  private ActivityType type;
 
-    By userLinkSelector = By.cssSelector("span.subtle > a");
-    By titleSelector = By.cssSelector("a.title");
-    userLink = parentElement.findElement(userLinkSelector);
-    titleLink = parentElement.findElement(titleSelector);
+  public Activity(WebElement activityWebElement) {
+    this.entry = activityWebElement;
+    this.type = getTypeFromEntry();
   }
 
-  public String getTitle() {
-    return titleLink.getText();
+  private ActivityType getTypeFromEntry() {
+    String cssClass = entry.getAttribute(HTML.Attribute.CLASS.toString());
+    return Arrays.stream(ActivityType.values())
+      .filter(type -> cssClass.contains(type.getCssType()))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "type" which hides the field declared at line 27. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AHiddenFieldCheck)